gradle/component.gradle

```diff
@@ -184,27 +220,24 @@ dependencies {
 
 task iosTest {
     def device = project.findProperty("iosDevice")?.toString() ?: "iPhone 8"
-    dependsOn 'linkDebugTestIosX64'
+    dependsOn 'linkDebugTestIos'
     group = JavaBasePlugin.VERIFICATION_GROUP
     description = "Runs tests for target 'ios' on an iOS simulator"
 
     doLast {
-        def binary = kotlin.targets.iosX64.binaries.getTest('DEBUG').outputFile
+        def binary = kotlin.targets.ios.binaries.getTest('DEBUG').outputFile
         exec {
             commandLine 'xcrun', 'simctl', 'spawn', device, binary.absolutePath
         }
     }
 }
 
-publishing {
-    publications {
-        kotlinMultiplatform {
-            artifactId = module.name
-        }
-    }
-}
+// uncomment to print all available configurations, useful to debug sourceSets and dependencies
+//configurations.each { println it }
 
-apply from: "../gradle/jacoco.gradle"
+if ((new File("../jacoco.gradle")).exists())
+    apply from: "../jacoco.gradle"
 
-apply from: "../gradle/publish.gradle"
+if ((new File("../publish.gradle")).exists())
+    apply from: "publish.gradle"
```

Comment:
File path for `if` check and for `apply` are different. `../publish.gradle` vs `publish.gradle`

Reply:
This is actually "works", `File()` has different root than `apply`. It will apply for the main modules (like `Components` or `logging`) and not for `example`.
Let's do these things:
- clean this up to do this better, like setting a flag to apply (if this is possible) or wrapping `component` into `publishablecomponent` or something where it applies components and then publish.
- add the `publishToMavenLocal` task to CI to make sure it does not break
- make sure Jacoco can actually also be applied for example, and not just the main modules

Reply:
I've done these, except applying jacoco to the example, on further consideration, code coverage there is not so important at the moment.